### ISSUE TITLE
Unable to Modify Table / Add Records if Default value is .5

### ISSUE THREAD

**Issue by abvanliew** [sqlitebrowser/sqlitebrowser#623]
I had created a table and used a default value of .5 for a REAL column. When I went to the Browse Data tab and selected the table, no column names or filter fields were shown. When I tried to Add Record I get "Error adding record: no such table: (INSERT INTO '' DEFAULT VALUES;)". Then I tried to modify the table. It brought up an empty Edit table definition window with none of my existing fields. After recreating the table I discovered that if I used a default value of 0.5 I did not see this problem.
![sqlite db bowser - modifying table](https://cloud.githubusercontent.com/assets/619227/16016510/5eaaa70e-3169-11e6-8748-c6ad69fd0db5.jpg)
![sqlite db bowser - adding record](https://cloud.githubusercontent.com/assets/619227/16016509/5e155cd0-3169-11e6-9fa8-0c1c6fb69f0a.jpg)

**Comment by chrisjlocke** [comment]
Not only that, but if you add a record via the 'Execute SQL' tab, the record displays in the 'Browse Data' tab.  Hover the mouse over the row, and DB Browser weeps, curls up in a ball and cries horribly.

![image](https://cloud.githubusercontent.com/assets/7872218/16045417/2cc9367e-3240-11e6-979e-001458500624.png)

**Comment by justinclift** [comment]
This definitely sounds like something we should track down and fix prior to 3.9.0.

**Comment by revolter** [comment]
@abvanliew, Changed the milestone for the release following the next one as it probably requires some major changes which I and @justinclift are not able to do for 3.9.0.

**Comment by innermous** [comment]
Then I take this, been lazy for last two weeks.

**Comment by revolter** [comment]
@innermous, I _think_ the problem is in the parser stuff (eg `grammar/*`). I thought that maybe the bug is in antlr and we should update it (though we can't, atm).

**Comment by MKleusberg** [comment]
Yep, the bug was in the ```grammar/*``` code, though not in the antlr code but in our grammar file which is then parsed by antlr. Should be fixed now :) 

@abvanliew Can you test the next nightly build and report back if the issue is fixed for you?

**Comment by justinclift** [comment]
@abvanliew Out of curiosity, would you have time to check if the fix @Martin added works ok?

We have an upcoming release (3.9.0) this weekend, that would be good to get the fix into (if ok).

I can create a Windows build for you to test with, if that helps. :smile:

**Comment by abvanliew** [comment]
Yeah if you can get me a windows build I would be more than happy to test it.

**Comment by justinclift** [comment]
@abvanliew Yep, can do.  Windows can run as either 32-bit, or 64-bit.  Do you know which one yours is?

If you're not sure, I'll just make a build for each.  It's not much extra time to do. :smile:

**Comment by MvGulik** [comment]
A Windows 32-bit for me please.
I think I have a similar/related crash case.

**Comment by justinclift** [comment]
@MvGulik Yep, not a problem. :smile:

It's late here at the moment (I need sleep now), but I'll get into this tomorrow morning.

In the meantime if you need a possible solution, try the automatically generated nightly build linked from here:

&nbsp; &nbsp; https://github.com/sqlitebrowser/sqlitebrowser#windows

That's a 32-bit one, and has the fix in it.  It also has some recent (untested experimental) code in it for other things... so I'd say there's a 50/50 chance of it even being usable.  But.. if it does, it might be ok until tomorrow. (make a backup of your data first!) :smile:

**Comment by MvGulik** [comment]
No rush.

With Nightly build (17-Aug-2016 02:21) things seems partially fix.

Its not crashing anymore (*when letting the mouse hover to long over a data field*) :smile: 

Table edit still showing
```
CREATE TABLE `subject_data_tmp` (

);
```
on a table created with:
```
CREATE TABLE subject_data_tmp (pid INTEGER PRIMARY KEY ASC, pr_id INTEGER NOT NULL DEFAULT ASC)
```
But that one is in the pipeline if I got it right.

**Comment by chrisjlocke** [comment]
Using the same nightly, if I change the SQL to `CREATE TABLE subject_data_tmp (pid INTEGER PRIMARY KEY ASC, pr_id INTEGER NOT NULL DEFAULT .5)` it works as required.
I also created a text field with a default of .5 (technically '.5') and that worked as desired too - it didn't get expanded to a default of 0.5.

**Comment by MvGulik** [comment]
Yea, that ```DEFAULT ASC``` was the trigger of the problems on this side (*an edit error from me that slipped by*). After changing it to a more appropriate and intended ```DEFAULT 0```. sqlitebrowser (*the nightly and the last final version*) went back to behaving as expected/normal.

(*Kept a copy of the original error case database for re-testing, just in case. If needed.*)

**Comment by justinclift** [comment]
Thanks guys, that's awesome. :smile:

I'll mark this as closed now, plus add the fix to the 3.9.x series code.

Official 3.9.0 will very likely be released this weekend, so it's not a long wait. :smile:
